README.md

```diff
@@ -53,7 +53,47 @@ cd onyo
 pip install -e ".[tests, docs]"
 ```
 
-### Tests
+### Design
+
+Onyo follows a set of development and coding conventions throughout its code
+base. For linting and type checking, the Onyo project uses flake8 and Pyre. The
+following sections describe additional conventions followed in this project,
+extending the existing [PEP 8 Style Guide](https://peps.python.org/pep-0008/).
+
+#### The Complete Code Base:
+- All classes, functions and properties have a doc-string describing their
+  functionality.
+- All functions have type hinting (including explicit return type for `None`).
+- The code conforms to [PEP 8](https://peps.python.org/pep-0008/).
+
+#### lib Tests (`tests/lib/`):
+- follow Onyo's code conventions, see [The Complete Code Base](#the-complete-code-base).
+- define the expected behavior of the code for all use-cases of the different
+  consumers (i.e. CLI, demos).
+- all public functions and properties of all classes are tested with all
+  possible combinations of input:
+  - single valid inputs, including different data-types (if allowed)
+  - multiple valid inputs (if allowed)
+  - without input (if allowed)
+  - one test for each possible error
+  - one test for each possible conflict with other inputs
+
+#### CLI Tests (`tests/commands/`):
+- follow Onyo's code conventions, see [The Complete Code Base](#the-complete-code-base).
+- define the expected behavior which the user can expect from Onyo.
```

Comment:
nit: these are sentence fragments (like above), and thus should not have a period at the end